crates/codegen/grammar/src/parser_definition.rs

```diff
@@ -40,16 +40,16 @@ pub enum ParserDefinitionNode {
     Versioned(Box<Self>, Vec<VersionQualityRange>),
     Optional(Box<Self>),
     ZeroOrMore(Box<Self>),
-    OneOrMore(Box<Self>),
-    Sequence(Vec<Self>),
-    Choice(Vec<Self>),
+    OneOrMore(String, Box<Self>),
+    Sequence(Vec<(String, Self)>),
+    Choice(String, Vec<Self>),
```

Comment:
I wonder how does this handle unnamed nodes, over using `Option<String>`?

Reply:
In the future, I planned to have `enum FieldName` that has a `None = 0` variant, to simplify the ergonomics and the matches; as such, I wanted to handle this as a single type rather than going through `Option<String>` and also because this would introduce two variants for empty names - `None` and `Some("")`.

This means that `""` value is treated as no name (technically, having `""` name doesn't make sense and is a bug) and is handled using `is_empty` checks to skip the naming of the node.

I think that in this case introducing an Option<String> may give a false idea of correctness but would be more unwieldy in terms of `Some`/`None` matching while not preventing having to do `is_empty` matches anyway.

Ideally, this should be solved with a custom type that ideologically looks like `Option<NonEmptyString>` but since I'm planning to have the flattened `FieldName` with `None = 0` anyway, I figured there's no point in doing that right now and would induce churn to introduce that only to immediately get rid of it in a follow-up.

Reply:
I think `Option` of enum more correctly models what it going on. And IMO it makes the code that handles it more obviously deal with a not-specified, whereas a `None` variant feels more like a default value.

My matching code presumes Option, and it's clearer IMO.

Reply:
Option of enum or flattened enum sounds good to me and is what I planned to do; I just wanted to avoid Option<String> here to reduce the churn. Do you want me to convert this to an Option<String> or just do the enum collection as part of this PR?

Reply:
Opened #713 that migrates to an `Option<FieldName>`, based on this branch.